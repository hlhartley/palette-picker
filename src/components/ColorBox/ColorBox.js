import React from 'react';

export const ColorBox = () => {
    return (
        <div className="ColorBox">
            <p>ColorBox</p>
        </div>
    )
}

export default ColorBox;