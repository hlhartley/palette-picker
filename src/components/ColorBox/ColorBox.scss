.ColorBox {
    border: 2px solid tomato;
}