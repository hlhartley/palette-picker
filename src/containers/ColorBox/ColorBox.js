import React from 'react';
import ColorCard from '../../containers/ColorCard/ColorCard';
import { connect } from 'react-redux';
import uuid from 'uuid/v4';

export const ColorBox = (props) => {
    console.log(props)
    return (
        <div className="ColorBox">
            {
                props.projects.map(project => {
                    return <ColorCard key={uuid()}/>
                })
            }
        </div>
    )
}

export const mapStateToProps = (state) => ({
    projects: state.projects
  });

export default connect(mapStateToProps)(ColorBox);