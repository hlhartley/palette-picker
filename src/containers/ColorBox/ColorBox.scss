.mapped-colors {
    border: 2px solid tomato;
    display: flex;
    flex-direction: column;
    height: 75vh;
}