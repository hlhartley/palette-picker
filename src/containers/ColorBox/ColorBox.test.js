import React from 'react';
import { shallow } from 'enzyme';
import { ColorBox, mapStateToProps } from './ColorBox';

describe('ColorBox', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <ColorBox hexcodes={['ffffff', 'fffff1', 'fffff2', 'fffff3', 'fffff4']}/>
      )
    });

  describe('ColorBox component', () => {
    it.skip('should properly render the component elements', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  // **Write returnColorCards and findCurrentPaletteColors tests here**

  describe('mapStateToProps', () => {
    it('should return an object with hexcodes as keys', () => {
      const mockState = {
        projects: [],
        isLoading: false,
        error: '',
        palettes: [],
        extra: '',
        lockedHexcodes: [],
        currentPaletteId: 1,
        hexcodes: [],
        shouldDisplayRandom: true
      }
      const expected = {
        hexcodes: []
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    });
  });
});