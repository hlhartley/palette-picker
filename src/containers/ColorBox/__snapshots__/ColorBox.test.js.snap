// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBox ColorBox component should properly render the component elements 1`] = `undefined`;
